xdsl/printer.py

```diff
@@ -119,6 +119,7 @@ def print(self, *argv: Any) -> None:
                 self.print_string(arg)
                 continue
             if isinstance(arg, SSAValue):
+                arg = cast(SSAValue[Attribute], arg)
```

Comment:
```suggestion
                _arg: SSAValue[Attribute] = arg
```

Reply:
I'm so confused on why this works in the first place actually.
This is technically a cast, right, why does this work?
I'm almost certain this behavior is a bug, as doing `_arg: SSAValue[IntegerType] = arg` works as well.

Reply:
Yeah seems like a bug indeed